contributors/design-proposals/sig-cli/kubectl-service-catalog.md

```diff
@@ -0,0 +1,227 @@
+# MVP - kubectl commands for the Service Catalog
+
+Status: Pending
+
+## Motivation
+
+The client binary for the service catalog must cover every [use case](use-cases.md) and provide 
+tools for every [persona](use-cases.md#personas) described. 
+
+While in some use cases we will most likely rely on the base commands already present on `kubectl` - like 
+`create`, `get`, `describe`, and `delete` - in others we want to have subcommands dedicated 
+to interacting with the service catalog. It is expected that the "end-users" (the *Service Consumer* 
+persona, for example), have a set of subcommands that are simple to use and allow them to easily check 
+information about the services available on a catalog, request instances, bind them to apps, and 
+manage their installed instances in a simple and user-friendly way.
+
+This document proposes the way every use case is fulfilled by `kubectl` subcommands and/or extensions.
+
+## Proposal
+
+Below the [identified personas](use-cases.md#personas) and the commands that are expected to fulfill the use
+cases for each one of them.
+
+Note that when relying on existing commands, some improvements to printers and describers of the involved 
+resources might be needed. For example, in the describer of a service instance we might want to display the 
+number of bindings that currently exists and other information useful to catalog operators.
+
+### Service consumer
+
+For this persona we want simple and user-friendly commands. Service consumers must be able to easily search the entire
+catalog for services available, check the metadata for each service, request an instance of a given service and
+bind it to apps, only through the command line without the need of json or yaml files. 
+
+So the proposal is to have a new set of subcommands called
+
+```
+kubectl catalog
+```
+
+which could be aliased as `kubectl (catalog | servicecatalog | service-catalog)` and distributed as part of the main 
```

Comment:
three initial categories of additions make sense to me:
* get/describe for custom objects accomplished server-side - https://github.com/kubernetes/community/pull/363
* generators for object-specific creation nested under `kubectl create <mykind>` that support all the conventions of the other `kubectl create` generators (--dry-run, different --output formats, etc). This seems like it should a really structured plugin (plugin can be interrogated for args/flags/usage/help info, user invokes kubectl, kubectl invokes plugin with args/flags, plugin outputs yaml or json that kubectl can transform/create/output in all the standard ways)
* completely independent commands that nest within kubectl